clients/src/main/resources/common/message/AddPartitionsToTxnRequest.json

```diff
@@ -23,17 +23,35 @@
   // Version 2 adds the support for new error code PRODUCER_FENCED.
   //
   // Version 3 enables flexible versions.
-  "validVersions": "0-3",
+  //
+  // Version 4 adds VerifyOnly field to check if partitions are already in transaction and adds support to batch multiple transactions.
+  "validVersions": "0-4",
   "flexibleVersions": "3+",
   "fields": [
-    { "name": "TransactionalId", "type": "string", "versions": "0+", "entityType": "transactionalId",
+    { "name": "VerifyOnly", "type": "bool", "versions": "4+", "default": false,
```

Comment:
I was wondering if having `VerifyOnly` as top level field is the right thing to do. It basically means that we cannot batch transactions with `VerifyOnly=true` and transactions with `VerifyOnly=false` in the same request. This may impact out batching mechanism. Thoughts?

Reply:
Is the idea that we will have some old producers (verifyOnly) and some new ones?

Reply:
One benefit is that verifyOnly requests won't be slowed down by the non-verify only ones

Reply:
Right. At the broker level, I think that it is possible to have old and new producers, right? If we use one connection from the broker to the transaction coordinator, it means that one has to wait on the other. I don't know how you plan to implement this so it is just a thought.

Reply:
It's a fair point. I'll think on it a bit.

Reply:
Yeah, interesting point. I could see moving `VerifyOnly` to the level of `transactionalId`. An interesting corollary if we do batch both modes together is that verify-only requests may end up blocking on replication even though we are only checking the state in memory. This would kind of penalize old clients, but maybe that's acceptable. Unless we used separate connections for each mode, perhaps it is unavoidable.

Reply:
Yup that was my point here:
> One benefit is that verifyOnly requests won't be slowed down by the non-verify only ones
But yeah, I guess we slow things down as well by using separate requests.

Reply:
Yep, you got there first 😄. I would probably vote to make the change. Tracking separate connections does not sound attractive. We could still do that even with the modified protocol if there is a noticeable regression for old clients.

Reply:
I'm wondering do we want to keep the top level verifyOnly as an optimization? Or is it just more trouble than it's worth and maybe won't be used too often.

Reply:
Actually -- the way this is implemented, we check on a per-transaction basis anyway. Since this is 1:1 corresponded to producer ID (and therefore client version) the only time we could see changes between verify only and not would be when the client upgrades. I may need to think about handling that as a special case.